pkgs/applications/networking/flood/default.nix

```diff
@@ -0,0 +1,17 @@
+{ pkgs, nodejs, nodePackages, stdenv, lib, ... }:
+
+let
+  nodePackagesLocal = import ./node-composition.nix {
```

Comment:
Can this reuse the shared node packages in nixpkgs? 9000 lines for one package is a bit much.

Reply:
@timokau I based this on other node packages that are applications too, like this one: https://github.com/NixOS/nixpkgs/blob/e7fe577d9803885d1191c6612b95c246cb605dde/pkgs/development/web/newman/node-packages.nix. There is other examples in nixpkgs too for node packages that brings its full dependency tree.

Why I preferred this approach in this package is that like many things in node ecosystem this package dependencies are always changing. Current master already have a completely different set of dependencies.

Reply:
> like many things in node ecosystem this package dependencies are always changing

But if you were to just add `flood` to the shared node packages, `generate.sh` should just add whatever dependencies are required by `flood` specifically, while using shared dependencies if available, no?

I'm not actually very familiar with the Node part of Nixpkgs either, I just did this one PR where I merged a lot of Node PRs with merge conflicts. But here I see no reason why `flood` can't be handled like all other Node tools.

Reply:
For context: All the node package definitions actually increase the repository size over time since they get regenerated a lot and each version is kept in git history.